contracts/VoteLockerCurve.sol

```diff
@@ -50,6 +56,13 @@ contract VoteLockerCurve {
     uint256 public constant MAX_LOCK_TIME = 4 * 365 * 86400; // 4 years
     /// @notice Vote boost if locked for maximum duration
     uint256 public constant MAX_VOTE_MULTIPLE = 4;
+    /// @notice Maximum number of delegators per account
+    uint256 private constant MAX_DELEGATORS = 100;
```

Comment:
We need to write tests for this. Although the views won't have an issue, the functions that call balanceOf and balanceOfAt that write state (e.g. the governance castVote is probably one) could become unusable if a full 100 delegators means that the function uses too much gas.

We need to try and figure out that most gassy function that calls the views that have the delegation loops, and write a test for the very worst case scenario. Even if the function is still callable we may need to reduce this number because we don't want voting for someone with a lot of delegations to cost a fortune.

Reply:
Thanks for pointing this out. After some testing I hate my implementation... So the bottom line: 
- each delegate adds ~50k gas to the `castVote` call
- can't test with more than 8 delegates using brownie+hardhat (even standalone node + brownie console in separate terminal) because it crashes hardhat. Brownie calls `debug_traceTransaction` [after tx execution](https://github.com/eth-brownie/brownie/blob/3aecd87f47c9c316c85b0b0c6252ff7d900cca74/brownie/network/transaction.py#L634) and hardhat [rolls over and dies](https://capture.dropbox.com/MJ4QCk5aEH9ArF9x) since the JSON it is trying to stringify is too large.

So I have reduces the max amount of delegators to 8. Obviously this approach has huge issues.

Reply:
I wonder if we should reduce all the checkpoints (voting graphs) of all delegators to one structure - similar to global checkpoints. But I fear that will put some heavy complexity to the contract code.

Need to think more about this one...